### ISSUE TITLE
[packager] Custom download strategy for homebrew

### ISSUE THREAD

**Issue by aalmiray** [jreleaser/jreleaser#1555]
### Discussed in https://github.com/jreleaser/jreleaser/discussions/1554

<div type='discussions-op-text'>

<sup>Originally posted by **jonnybbb** December 15, 2023</sup>
I would like to use a custom download strategy for downloading the artefacts. 

```
require_relative "lib/private"
.....
  if OS.linux? && Hardware::CPU.intel?
    URL "<url>", using: GitHubPrivateRepositoryReleaseDownloadStrategy
    sha256 "<sha>"
  end
  if OS.mac? && Hardware::CPU.arm?
    url "<url>", using: GitHubPrivateRepositoryReleaseDownloadStrategy
    sha256 "<sha>"
  end
```
Is there a way to configure this in jReleaser? 

goreleaser [has this](https://goreleaser.com/customization/homebrew/):
```
    # Allows you to set a custom download strategy. Note that you'll need
    # to implement the strategy and add it to your tap repository.
    # Example: https://docs.brew.sh/Formula-Cookbook#specifying-the-download-strategy-explicitly
    download_strategy: CurlDownloadStrategy

    # Allows you to add a custom require_relative at the top of the formula
    # template.
    custom_require: custom_download_strategy
```

**Comment by aalmiray** [comment]
You can try this feature using the `early-access` builds of the CLI.

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.10.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.10.0))
